### ISSUE TITLE
Add code from the mitiq paper

### ISSUE THREAD

**Issue by nathanshammah** [unitaryfund/mitiq#775]
The co-authors have agreed on making public the code relevant to the code snippets and plots appearing figures of the Mitiq paper (https://arxiv.org/abs/2009.04417). 

We discussed different possible routes. One was to have a separate repository, `mitiq-paper`. The one we are implementing, to ensure that the code keeps working and to reduce overhead, is to have the code snippets and plots included in the documentation of the library (run and tested with continuous integration).

We proposed to have the jupyter notebooks along with other examples in the **Mitiq Examples** section of the Users Guide in the documentation. 

We will also have accompanying data, that I propose we include simply in a `data/` subfolder there, under `docs/source/examples`. It is not heavy so it shouldn't be worth to selectively exclude it when cloning the repository. 

We also mentioned the fact that tags for the arXiv versions could help us pinpoint eventual modifications in the repository. 

I propose to also add some information about it under the **Research** section of the documentation. 

We did not create a Zenodo repository to avoid duplications of bibliographic records, with respect to the paper (once published) as this generates a DOI.

**Comment by crazy4pi314** [comment]
Copying over notes from duplicate issue:

The current code for figures and data for the Mitiq paper is a bit spread out and as discussed a the 6/25 community call we decided to add all data/sample code/figure code to the docs.

- [ ] Setup directory in the docs for data
- [ ] Add what raw data we have
- [ ] Add current plot source
- [ ] Add or link to any example code shown
- [ ] Setup testing for any samples if needed
- [ ] Tag the commit when merged

**Comment by nathanshammah** [comment]
We discussed this at the Mitiq meeting today. 

We decided the following: 
- The **data** is not large (less than 1MB) and it is most straightforward to include it here, for coherence and future maintenance of the related notebooks. 
- I will downscope #776 to include only the jupyter notebook with the **code snippets** and a folder with the raw data, so that this action item can be ticked by the ~time we make the arXiv v2 update for the Mitiq paper. 
- We will open **three separate issues** for notebooks related to the three figures to be included in the **Examples** of the documentation. We will keep the focus on the subject matter and cursorily mention at the beginning that these are also supporting the paper: 
- 1. ZNE on IBMQ and Rigetti; 
- 2. ZNE on VQE for H2 energies 
- 3. PEC simulation.  
- The place to reference to the code snippets, data and related notebooks will be in the **Research** section of the documentaion.
